julia/src/BridgeStan.jl

```diff
@@ -0,0 +1,336 @@
+
+function param_unc_names(sm::StanModel)
+    cstr = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_unc_names"),
+        Cstring,
+        (Ptr{StanModelStruct},),
+        sm.stanmodel,
+    )
+    [string(s) for s in split(unsafe_string(cstr), ',')]
+end
+
+function param_constrain!(
+    sm::StanModel,
+    theta_unc,
+    out::Vector{Float64};
+    include_tp = false,
+    include_gq = false,
+)
+    dims = param_num(sm; include_tp = include_tp, include_gq = include_gq)
+    if length(out) != dims
+        throw(
+            DimensionMismatch("out must be same size as number of constrained parameters"),
+        )
+    end
+    rc = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_constrain"),
+        Cint,
+        (Ptr{StanModelStruct}, Cint, Cint, Ref{Cdouble}, Ref{Cdouble}),
+        sm.stanmodel,
+        include_tp,
+        include_gq,
+        theta_unc,
+        out,
+    )
+    if rc != 0
+        error("param_constrain failed on C++ side; see stderr for messages")
+    else
+        out
+    end
+end
+
+function param_constrain(sm::StanModel, theta_unc; include_tp = false, include_gq = false)
+    out = zeros(param_num(sm, include_tp = include_tp, include_gq = include_gq))
+    param_constrain!(sm, theta_unc, out; include_tp = include_tp, include_gq = include_gq)
+end
+
+
+function param_unconstrain!(sm::StanModel, theta, out::Vector{Float64})
+    dims = param_unc_num(sm)
+    if length(out) != dims
+        throw(
+            DimensionMismatch(
+                "out must be same size as number of unconstrained parameters",
+            ),
+        )
+    end
+
+    rc = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_unconstrain"),
+        Cint,
+        (Ptr{StanModelStruct}, Ref{Cdouble}, Ref{Cdouble}),
+        sm.stanmodel,
+        theta,
+        out,
+    )
+    if rc != 0
+        error("param_unconstrain failed on C++ side; see stderr for messages")
+    else
+        out
+    end
+end
+
+function param_unconstrain(sm::StanModel, theta)
+    out = zeros(param_unc_num(sm))
+    param_unconstrain!(sm, theta, out)
+end
+
+function param_unconstrain_json!(sm::StanModel, theta::String, out::Vector{Float64})
+    dims = param_unc_num(sm)
+    if length(out) != dims
+        throw(
+            DimensionMismatch(
+                "out must be same size as number of unconstrained parameters",
+            ),
+        )
+    end
+
+    rc = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_unconstrain_json"),
+        Cint,
+        (Ptr{StanModelStruct}, Cstring, Ref{Cdouble}),
+        sm.stanmodel,
+        theta,
+        out,
+    )
+    if rc != 0
+        error("param_unconstrain_json failed on C++ side; see stderr for messages")
+    else
+        out
+    end
+end
+
+function param_unconstrain_json(sm::StanModel, theta::String)
+    out = zeros(param_unc_num(sm))
+    param_unconstrain_json!(sm, theta, out)
+end
+
+function log_density(sm::StanModel, q; propto = true, jacobian = true)
+    lp = Ref{Float64}(0.0)
```

Comment:
```suggestion
    lp = Ref(0.0)
```
The number `0.0` makes the type `Float64` implicitly.